pysmt/rewritings.py

```diff
@@ -618,26 +618,78 @@ def walk_ite(self, formula, args, **kwargs):
         else:
             return formula
 
+
+class TimesDistributor(IdentityDagWalker):
+    """Normalize the use of multiplication by pushing it into the leafs.
+
+    E.g., (x+1)*3 -> (x*3) + 3
```

Comment:
will this rewriting handle the case when both lhs and rhs are plus or minus nodes.

for example: 
```
(x+1) * (x -1) -> x*x -1*x + 1*x - 1*1
```

I am afraid, it will not. 

What do you think?